server/src/main/java/org/opensearch/tracing/OTelResourceProvider.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.tracing;
+
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.trace.propagation.W3CTraceContextPropagator;
+import io.opentelemetry.context.propagation.ContextPropagators;
+import io.opentelemetry.exporter.otlp.trace.OtlpGrpcSpanExporter;
+import io.opentelemetry.sdk.OpenTelemetrySdk;
+import io.opentelemetry.sdk.resources.Resource;
+import io.opentelemetry.sdk.trace.SdkTracerProvider;
+import io.opentelemetry.sdk.trace.export.BatchSpanProcessor;
+import io.opentelemetry.semconv.resource.attributes.ResourceAttributes;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * This class encapsulates all OpenTelemetry related resources
+ */
+class OTelResourceProvider {
+
+    private static final Resource resource;
+    private static final ContextPropagators contextPropagators;
+    private static volatile OpenTelemetry OPEN_TELEMETRY;
+
+    static {
+        resource = Resource.getDefault().merge(Resource.create(Attributes.of(ResourceAttributes.SERVICE_NAME, "OpenSearch")));
+        contextPropagators = ContextPropagators.create(W3CTraceContextPropagator.getInstance());
+    }
+
+    static OpenTelemetry getOrCreateOpenTelemetryInstance(TracerSettings tracerSettings) {
```

Comment:
All the static context would be initialized even when tracing is disabled? what is the overhead?

Reply:
Have moved the `Resource` inside method call. Also loading `ContextPropagators` would not add overhead as its a lightweight class. 

Further `OTelResourceProvider.java` should be loaded only when referenced i.e when tracing is enabled.